### PR TITLE
Cleanup Dockerfile

### PR DESCRIPTION
**Pull request by SgtSilvio**
https://hivemq.kanbanize.com/ctrl_board/41/cards/18943/details/

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,41 +1,39 @@
...
 
-COPY hivemq-ce-${HIVEMQ_VERSION}.zip /
-RUN unzip /hivemq-ce-${HIVEMQ_VERSION}.zip -d /opt \
-    && chgrp -R 0 /opt \
-    && chmod -R 770 /opt
+COPY hivemq-ce-${HIVEMQ_VERSION}.zip /tmp/hivemq-ce.zip
+RUN unzip /tmp/hivemq-ce.zip -d /opt \
+    && mv /opt/hivemq-ce-${HIVEMQ_VERSION} /opt/hivemq
+COPY config.xml /opt/hivemq/conf/config.xml
+RUN chmod -R 770 /opt \
```
Comment (LukasBrand) [inline docker/Dockerfile:10]: So we want to keep rwx permissions for the complete opt folder?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
+    && chmod +x /opt/hivemq/bin/run.sh
```
Comment (LukasBrand) [inline docker/Dockerfile:11]: Isn't his covered by the statement above?

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,41 +1,39 @@
-# We use multi-stage here to unzip in an initial layer so we don't have to COPY and then RUN unzip (two layers). ADD can lead to larger layers as well.
-FROM busybox:1.35.0@sha256:02289a9972c5024cd2f083221f6903786e7f4cb4a9a9696f665d20dd6892e5d6 AS unpack
+# Additional build image to unpack the zip file and change the permissions without retaining large layers just for those operations
```
Comment (LukasBrand) [inline docker/Dockerfile:1]: Retaining large layers means bloating the image? 

I like the improved doc. Just a suggestion: Wouldn't it be a bit more explanatory if there is sth like:
```
# Image 1: Additional build image to

# Image 2: Actual ...
```
Just to kind of separate the logic of the file into two sections

File: docker/Dockerfile [hunk 4]
```diff
@@ -1,41 +1,39 @@
...
+RUN unzip /tmp/hivemq-ce.zip -d /opt \
+    && mv /opt/hivemq-ce-${HIVEMQ_VERSION} /opt/hivemq
+COPY config.xml /opt/hivemq/conf/config.xml
+RUN chmod -R 770 /opt \
+    && chmod +x /opt/hivemq/bin/run.sh
+
 
 FROM eclipse-temurin:11.0.22_7-jre-jammy@sha256:985ec5f9ff61ef9e24b2298ef902d773a1e1cb36693d62d0ac4d37e289ff595b
 
 ARG HIVEMQ_VERSION
```
Comment (LukasBrand) [inline docker/Dockerfile:16]: This is not used anymore, is it?

Comment (afalhambra-hivemq) [reply docker/Dockerfile:16]: +1

**Review by afalhambra-hivemq — APPROVED** [review]

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,41 +1,39 @@
...
+COPY --from=unpack /opt/hivemq /opt/hivemq
+RUN chmod +x /opt/docker-entrypoint.sh \
     && groupadd --gid ${HIVEMQ_GID} hivemq \
     && useradd -g hivemq -d /opt/hivemq -s /bin/bash --uid ${HIVEMQ_UID} hivemq \
-    && chgrp 0 /opt/hivemq-ce-${HIVEMQ_VERSION}/conf/config.xml \
-    && chmod 770 /opt/hivemq-ce-${HIVEMQ_VERSION}/conf/config.xml \
-    && chgrp 0 /opt/hivemq \
-    && chmod 770 /opt/hivemq \
-    && chmod +x /opt/hivemq/bin/run.sh /opt/docker-entrypoint.sh
+    && chmod 770 /opt/hivemq
```
Comment (LukasBrand) [inline docker/Dockerfile:36]: Shouldn't hivemq own this directory as well? Or is this implicitly done?

File: docker/Dockerfile (hunk 1 above)
Comment (SgtSilvio) [reply docker/Dockerfile:10]: Will change it to `/opt/hivemq`.

File: docker/Dockerfile (hunk 5 above)
Comment (SgtSilvio) [reply docker/Dockerfile:34]: no, currently this is owned by root and group 0

File: docker/Dockerfile (hunk 2 above)
Comment (SgtSilvio) [reply docker/Dockerfile:11]: This was not changed. I plan to change permissions in a follow up.

File: docker/Dockerfile (hunk 3 above)
Comment (SgtSilvio) [reply docker/Dockerfile:1]: done

**Review by ymengesha — APPROVED** [review]

**Review by LukasBrand — APPROVED** [review]

**Pull request merged**